bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/library/types/DecimalType.java

```diff
@@ -165,6 +165,9 @@ public State as(Class<? extends State> target) {
             } else {
                 return UnDefType.UNDEF;
             }
+        } else if (target == HSBType.class) {
+            return new HSBType(DecimalType.ZERO, PercentType.ZERO,
+                    new PercentType(this.toBigDecimal().multiply(new BigDecimal(100))));

```

Comment:
An alternative approach to minimizing the number of duplicate immutable objects in the heap, but with more certainty of outcome, would be to define and use a static member HUNDRED like
```
private static final BigDecimal HUNDRED = BigDecimal.valueOf(100);
```